src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,124 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_LENGTH = 10;
+    @SuppressWarnings("unchecked")
+    private T[] array = (T[]) new Object[INITIAL_LENGTH];
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == array.length) {
+            grow();
+        }
```

Comment:
you call grow method only inside such `if` blocks
include this condition inside method
you can rename it to represent condition ('checkSizeAndGrow', 'growIfRequied' etc.)